syntax = "proto3";
package config.conf;

message Server {
  string host = 1;
  uint32 port = 2;
}

message Database {
  string source = 1;
  uint32 pool_timeout = 2;
  uint32 pool_size = 3;
  uint32 pool_recycle = 4;
  uint32 connect_timeout = 5;
}

message JWT {
  string secret_key = 1;
  uint32 expiration_time = 2;
}

message ObjectStorage {
  string endpoint = 1;
  string access_key = 2;
  string secret_key = 3;
}

message Bootstrap {
  Server server = 1;
  Database database = 2;
  JWT jwt = 3;
  ObjectStorage oss = 4;
}
